libs/ui-toolkit/src/components/show-more/show-more.tsx

```diff
@@ -0,0 +1,78 @@
+import classNames from 'classnames';
+import { useRef, useState, useEffect } from 'react';
+import { t } from '@vegaprotocol/i18n';
+import { Button } from '../button';
+import type { ReactNode } from 'react';
+
+type ShowMoreProps = {
+  children: ReactNode;
+  closedMaxHeightPx?: number;
+  overlayColourOverrides?: string;
+};
+
+export const ShowMore = ({
+  children,
+  closedMaxHeightPx = 125,
+  overlayColourOverrides,
+}: ShowMoreProps) => {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const [expanded, setExpanded] = useState(false);
+
+  useEffect(() => {
+    const checkHeight = () => {
+      const container = containerRef.current;
+      if (container) {
+        container.scrollHeight < closedMaxHeightPx
+          ? setExpanded(true)
+          : setExpanded(false);
+      }
+    };
+
+    checkHeight();
+
+    window.addEventListener('resize', checkHeight);
+
+    return () => {
+      window.removeEventListener('resize', checkHeight);
+    };
+  }, [closedMaxHeightPx]);
+
+  const containerClasses = classNames(
+    'overflow-hidden transition-all ease-in-out duration-300',
+    {
+      'max-h-none': expanded,
+    }
+  );
+
+  const overlayClasses = classNames(
+    `absolute w-full h-16 bottom-0 left-0 transition-opacity duration-300 bg-gradient-to-b from-transparent ${
+      overlayColourOverrides ? overlayColourOverrides : 'to-white dark:to-black'
+    }`,
+    {
+      hidden: expanded,
+    }
+  );
+
+  return (
+    <>
+      <div className="relative">
+        <div
+          ref={containerRef}
+          className={containerClasses}
+          style={{ maxHeight: expanded ? 'none' : `${closedMaxHeightPx}px` }}
+        >
+          {children}
+        </div>
+        <div className={overlayClasses}></div>
+      </div>
+
+      {!expanded && (
```

Comment:
**thought (non-blocking)**: I hadn't read the ticket for this so the logic surprised me. I'm. fine with there being no 'collapse' button, but I wonder if we'll get feedback asking for one? Probably not. Perhaps we will from other apps that use this component. Easy enoguh to add when it's relevant.

<img width="1127" alt="Screenshot 2023-08-22 at 16 39 10" src="https://github.com/vegaprotocol/frontend-monorepo/assets/6678/e84e193a-fa60-4401-8503-3f4e24b57ef5">
<img width="1082" alt="Screenshot 2023-08-22 at 16 39 15" src="https://github.com/vegaprotocol/frontend-monorepo/assets/6678/a0f84e34-074c-4a01-9c20-9ae36374ec98">

Reply:
I ummed and ahhed about it. In the end, I decided to keep it simple, but you make a good point, I'll think some more.

Reply:
Based on our conversation, I'll keep it simple for now